Search/interpolation_search.cpp

```diff
@@ -0,0 +1,41 @@
+#include<iostream>
+using namespace std;
+
+int search(int arr[],int value,int len)		//function to search the value in an array using interpolation search
+{
+	int low=0,high,mid;
+	high=len-1;
+	while(arr[low]<=value and arr[high]>=value)
+		{
+			mid=(low + (( value-arr[low])*(high-low)) / (arr[high]-arr[low]));
+			if(arr[mid]>value)
+				high=mid-1;
+			else if (arr[mid]<value)
+				low=mid+1;
+			else
+				return mid; 
+		}
+	if(arr[low]== value)
+		return low;
+	return 0;
+}
+
+int main() 		//main function
+{
+	int n,value,array[100];
+	cout<<"Enter the size of array : ";
```

Comment:
please specify that the size of the array should be <=100 or dynamically allocate memory according to the input size.